modules/tinymce/src/core/main/ts/selection/DetailsElement.ts

```diff
@@ -11,6 +12,23 @@ const preventSummaryToggle = (editor: Editor): void => {
   });
 };
 
+const isHeading = (node: AstNode) => [ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ].includes(node.name);
```

Comment:
So move this to a new module like `AstNodeType`. Then use is in this module as a named import like `AstNodeType.isHeading` there are likely other candidates to become a generic predicate like that later on.